src/react-native/06-building-custom-components/building-custom-components.md

```diff
@@ -92,11 +181,67 @@ If you’ve implemented the solution correctly, the tests will pass when you run
 
 ## Objective 3: Organizing code: Modlets pattern
 
-TODO
+Our efforts to refactor have been going swimmingly; however, the `App.tsx` file is getting a bit crowded. It's about time we take the time to organize our code, while doing so we'll also introduce a useful pattern to follow.
+
+### Fractal Modlets
+
+Fractal Modlets, or just modlets, are a folder/file structure that place a heavy emphasis on the idea of abstraction. 
+
+They are completely self contained; each modlet is basically its own application. It has everything it needs inside of it.
+
+Each modlet is treated as a black box that can only be accessed through a single point: its index file.
+
+Modlets follow these three rules: 
+
+1. Only import folders not specific files 
+2. Do not reach “up” into parent directories or “sideways” into sibling directories to import things. Only “down” to child modlets, or to designated shared modlets in a parent.
+3. Each modlet is a “black box” that controls what it exports through its index file. Anything exported in the index should be stable: Don’t change those api’s unless you have to!
+
+These will make sense as we go along.
+
+### Modlets Example
+
+We’re building a simple product page for an e-commerce site. It has three components:
+
+- Product Details
+- Image Carousel
+- Add To Cart Button
+
+Without a plan of how to organize our files, we might end up with something like this:
+
+<img alt="Screenshot of an unorganized unplanned folder structure, yuck." src="../../static/img/react-native/06-custom-components/02-image.png" style="max-height: 750px; border: 4px solid black; border-radius: 25px;"/>
+
+The same page, refactored into a modlet architecture would look like this:
+
+<img alt="Screenshot of a folder structure that follows the standards of modlet architecture, neat." src="../../static/img/react-native/06-custom-components/03-image.png" style="max-height: 500px; max-width: 350px; border: 4px solid black; border-radius: 25px;"/>
```

Comment:
Do something like this instead: https://github.com/bitovi/academy/blob/e7126b74723910061d7c22e1bffb746cdcb43b57/src/react-vite/07-styling-in-react/styling-in-react.md#L126-L134

Preview: https://www.bitovi.com/academy/learn-react/styling-in-react.html#introducing-css-modules